ComponentKit/DataSources/CKCollectionViewTransactionalDataSource.mm

```diff
@@ -93,14 +95,65 @@ - (void)transactionalComponentDataSource:(CKTransactionalComponentDataSource *)d
                   didModifyPreviousState:(CKTransactionalComponentDataSourceState *)previousState
                        byApplyingChanges:(CKTransactionalComponentDataSourceAppliedChanges *)changes
 {
-  [_collectionView performBatchUpdates:^{
-    applyChangesToCollectionView(_collectionView, _attachController, [_componentDataSource state], changes);
-    // Detach all the component layouts for items being deleted
-    [self _detachComponentLayoutForRemovedItemsAtIndexPaths:[changes removedIndexPaths]
-                                                    inState:previousState];
-    // Update current state
-    _currentState = [_componentDataSource state];
-  } completion:NULL];
+  const BOOL changesIncludeNonUpdates = (changes.removedIndexPaths.count ||
+                                         changes.insertedIndexPaths.count ||
+                                         changes.movedIndexPaths.count ||
+                                         changes.insertedSections.count ||
+                                         changes.removedSections.count);
+  const BOOL changesIncludeOnlyUpdates = (!!changes.updatedIndexPaths.count && !changesIncludeNonUpdates);
+  
+  CKTransactionalComponentDataSourceState *state = [_componentDataSource state];
+  
+  if (changesIncludeOnlyUpdates) {
+    // We are not able to animate the updates individually, so we pick the
+    // first bounds animation with a non-zero duration.
+    CKComponentBoundsAnimation boundsAnimation = {};
+    for (NSIndexPath *indexPath in changes.updatedIndexPaths) {
+      boundsAnimation = [[state objectAtIndexPath:indexPath] boundsAnimation];
+      if (boundsAnimation.duration)
+        break;
+    }
+    
+    void (^updateState)(CKTransactionalComponentDataSourceState *) = ^(CKTransactionalComponentDataSourceState *state) {
```

Comment:
`state` should be renamed as it shadows the local variable declared on line 105.